android/library/src/main/java/com/polidea/multiplatformbleadapter/BleModule.java

```diff
@@ -908,11 +910,18 @@ public void action(BleError error) {
 
         Observable<RxBleConnection> connect = device
                 .establishConnection(autoConnect)
+                .doOnSubscribe(new Action0() {
+                    @Override
+                    public void call() {
+                        onConnectionStateChangedCallback.onEvent(ConnectionState.CONNECTING);
+                    }
+                })
                 .doOnUnsubscribe(new Action0() {
                     @Override
                     public void call() {
                         oneTimeErrorCallback.execute(BleErrorUtils.cancelled());
                         onDeviceDisconnected(device);
+                        onConnectionStateChangedCallback.onEvent(ConnectionState.DISCONNECTED);
```

Comment:
This event will be send twice because doOnUnsubscribe is also called right after doOnError.

Reply:
I don't think it's much of an issue really, but we can discuss it tomorrow in more details.